version: '3'

volumes:
  local_postgres_data: {}
  local_postgres_data_backups: {}
  local_redis_data: {}

services:
  django: &django
    build:
      context: .
      dockerfile: ./etc/Dockerfile
    image: meta_backend_local_django
    user: root
    depends_on:
      - postgres
      - redis
    volumes:
      - .:/app
    env_file:
      - ./.envs/.local/.django
      - ./.envs/.local/.postgres
      - ./.envs/.local/.secret
    ports:
      - "8080:8080"
    command: /bin/sh -c "python manage.py runserver_plus 0.0.0.0:8080"

  postgres:
    image: postgres:11-alpine
    volumes:
      - local_postgres_data:/var/lib/postgresql/data
      - local_postgres_data_backups:/backups
    env_file:
      - ./.envs/.local/.postgres
    ports:
      - "5435:5432"

  redis:
    image: redis:6.2-alpine
    volumes:
      - local_redis_data:/data

  celeryworker:
    <<: *django
    image: backend_meta_local_celeryworker
    depends_on:
      - redis
      - postgres
    volumes:
      - .:/app:z
    ports: [ ]
    command: celery -A config.celery worker -l INFO

  celerybeat:
    <<: *django
    image: backend_meta_local_celerybeat
    depends_on:
      - redis
      - postgres
    volumes:
      - .:/app:z
    ports: [ ]
    command: celery -A config.celery beat -l INFO
